### PR TITLE
Fix erroneous advancing of PC when setting one prog stmt

### PR DESCRIPTION
**Pull request by yanchith**
This is one small part extracted from the work on UI, so that that PR can be smaller when it lands.

### CODE REVIEW

File: src/interpreter/mod.rs [hunk 1]
```diff
@@ -323,8 +323,11 @@ impl Interpreter {
     }
 
     pub fn set_prog_stmt_at(&mut self, index: usize, stmt: ast::Stmt) {
-        // Mark the stmt and everything following it for re-execution
-        self.pc = index;
+        // Mark the stmt and everything following it for re-execution,
+        // if we already executed that far
+        if self.pc > index {
+            self.pc = index;
+        }
```
Comment (janper) [inline src/interpreter/mod.rs:330]: not that i'd think this is wrong, but is this better than `self.pc = min(self.pc, index);`?

**Review by janper — COMMENTED** [review]
I only have one question because i don;t understand the rest ;)

File: src/interpreter/mod.rs (hunk 1 above)
Comment (yanchith) [reply src/interpreter/mod.rs:330]: Just a matter of style

**Review by ondrowan — CHANGES_REQUESTED** [review]
The new test actually passes even with the old behaviour.

**Comment by yanchith** [comment]
@ondrowan oops, that was a copy paste error. Now it doesn't.

**Review by ondrowan — APPROVED** [review]

**Pull request merged**